Registry tutorial

gprashant17 opened the pull request:
Added registry tutorial in notebooks folder

mrsamsami committed:
Add env tutorial

mrsamsami committed:
Update env tutorial

gprashant17 committed:
Add files via upload

gprashant17 committed:
Update registry.ipynb

gprashant17 committed:
Update registry.ipynb

gprashant17 committed:
Update registry.ipynb

gprashant17 committed:
Update registry.ipynb

gprashant17 committed:
Update registry.ipynb

gprashant17 committed:
Update registry.ipynb

gprashant17 committed:
Update registry.ipynb

gprashant17 committed:
Update registry.ipynb

gprashant17 committed:
Update registry.ipynb

gprashant17 committed:
Update registry.ipynb

Prishruit commented on notebooks/registry.ipynb:
```diff
@@ -0,0 +1,615 @@
+      "cell_type": "markdown",
+      "source": [
+        "# About the tutorial\n",
+        "In this tutorial, we will explain how to register classes and objects into the RLHive framework."
```
Could you also add here why this is important and where it can be helpful for someone.

gprashant17 replied:
I moved the intro to registry content to the top. Hope that helps.

Prishruit commented on notebooks/registry.ipynb:
```diff
@@ -0,0 +1,615 @@
+        "Installation - To install a specific branch from github using pip - first copy the **clone URL** of the package and then add **@ symbol** with the name of the specific branch of the package that you want to install from. Example to install from the dev branch use : \\\\\n",
+        "`https://github.com/chandar-lab/RLHive.git@dev`"
+      ]
+    },
```
I do not think this is required in this tutorial, this part has been covered in other tutorials.

Prishruit commented on notebooks/registry.ipynb:
```diff
@@ -0,0 +1,615 @@
+        "!pip install pyglet\n",
+        "\n",
+        "!pip install git+https://github.com/chandar-lab/RLHive.git@dev"
+      ],
```
can use %%capture to prevent installation logs where the logs are too long in the notebook.

Prishruit commented on notebooks/registry.ipynb:
```diff
@@ -0,0 +1,615 @@
+        "from IPython import display as ipythondisplay\n",
+        "from hive.utils.utils import Registrable\n",
+        "import os\n",
+        "import sys"
```
Remove the imports that are not required or that are not used in this tutorial one ex is DQNAgent

Prishruit commented on notebooks/registry.ipynb:
```diff
@@ -0,0 +1,615 @@
+      "source": [
+        "!pip install gym==0.21.0\n",
+        "!pip install rlhive[atari]"
+      ],
```
Add %%capture to prevent installation logs

gprashant17 replied:
Added

Prishruit commented on notebooks/registry.ipynb:
```diff
@@ -0,0 +1,615 @@
+      "outputs": []
+    },
+    {
+      "cell_type": "markdown",
+      "source": [
+        "More than one environment can be registered at once using the `register_all` method. Consider registering three environments, `Gridv1`, `Gridv2`, and `Gridv3` in the RLHive registry."
+      ],
+      "metadata": {
+        "id": "sOy_VsfPLAIp"
+      }
+    },
```
I think a short description about the arguments of the register_all method would be helpful here.

Prishruit commented on notebooks/registry.ipynb:
```diff
@@ -0,0 +1,615 @@
+      }
+    },
+    {
+      "cell_type": "code",
+      "source": [
+        "registry.register_all(\n",
+        "    BaseEnv,\n",
+        "    {\n",
+        "        \"Gridv1\": Gridv1,\n",
+        "        \"Gridv2\": Gridv2,\n",
+        "        \"Gridv3\": Gridv3,\n",
+        "    },\n",
+        ")"
+      ],
```
This cell gives an error while executing "NameError: name 'Gridv1' is not defined", either define it or if you want to just give an example I think it would be better to put in a text cell. And the same goes for the following examples in the notebook.

Prishruit commented on notebooks/registry.ipynb:
```diff
@@ -0,0 +1,615 @@
+    {
+      "cell_type": "code",
+      "source": [
+        "registry.register_all(\n",
+        "    Agent,\n",
+        "    {\n",
```
I think there is a typo here the base class should be Logger instead of Agent in this case, let me know if I am wrong.

Prishruit reviewed (changes requested):
Please address the requested changes.

gprashant17 committed:
Made necessary changes to registry tutorial

Moved intro to top, removed installation instructions, removed unnecessary imports, added description of arguments for register and registrer_all, defined additional classes, corrected type

gprashant17 committed:
Update registry.ipynb

gprashant17 committed:
Update registry.ipynb

dapatil211 commented:
Can you please close this PR and create another one that only has the changes you made? ie doesn't include the env tutorial changes.
